### ISSUE TITLE
Make LeaveDiary aware of public/bank holidays

### ISSUE THREAD

**Issue by mattbostock** [mattbostock/timeoff#13]
Work out a way to make LeaveDiary aware of public or bank holidays.

Doesn't need to support public holidays for all countries. I imagine that this data will be used more as a hint for users rather than something that's strictly enforced, for these reasons:

a) Not everyone can take bank holidays as leave; they don't have to be included in the statutory leave allowance.

b) If the data has inaccuracies, users should be able to ignore it.

**Comment by ajlanghorn** [comment]
[This](https://www.gov.uk/bank-holidays.json) might come in handy here.

**Comment by mattbostock** [comment]
Thanks Andrew, that could be useful.

**Comment by mattbostock** [comment]
I've updated the description to make this issue clearer.

**Comment by mattbostock** [comment]
This is likely blocked until #15 is implemented.

**Comment by ajlanghorn** [comment]
I agree #15 needs doing before this. Thinking about it, I wonder if using the Google Calendar Public Holidays calendar in JSON form using the [Google Calendar API](https://developers.google.com/google-apps/calendar/v3/reference/) might be a better way - at least you'd be easily able to get public holidays from all around the world.

**Comment by mattbostock** [comment]
That's a good idea. We'd just need to be sure that we have the rights to use the data and work out a way to cache the data so we're updating it not more often than once every 24 hours.

The most important thing is that any failure to retrieve this data should not prevent a user from adding a leave request.
